backend/test/index.test.js

```diff
@@ -1,85 +1,127 @@
+    let createLobbyDTO = new CreateLobbyDTO("Anmol");
 
-        // Request to create a new lobby
-        clientSocket.emit('create-lobby', createLobbyDTO);
-    });
+    // Request to create a new lobby
+    // this will set the socket id to the host to be tested.
+    clientSocket.emit("create-lobby", createLobbyDTO);
 
-    test('Two hosts with two rooms', (done) => {
-        let createLobbyDTO = new CreateLobbyDTO('Anmol');
-
-        // Client 1 to subscribe to lobby-code
-        clientSocket.on('lobby-code', (lobbyCodeDTO) => {
-            expect(lobbyCodeDTO.code).toBeDefined();
-            // Wait for the other client to throw errors, if there are any.
-            setTimeout(function () {
-                done();
-            }, 1000);
-        });
-
-        let clientSocket2 = new Client(`http://localhost:4001`);
-        // Client 2 to subscribe to lobby-code
-        clientSocket2.on('lobby-code', (lobbyCodeDTOString) => {
-            throw new Error("Client 2 shouldn't receive a lobby code");
-        });
-
-        // Request to create a new lobby
-        clientSocket.emit('create-lobby', createLobbyDTO);
+    // Subscribe to game start event
+    // server should send this event back once host clicks start.
+    // Client to subscribe to "game-start"
+    clientSocket.on("game-start", () => {
+      done();
     });
 
-    test('Reset lobby', (done) => {
-        let createLobbyDTO = new CreateLobbyDTO('Anmol');
+    // Request to server that game is ready to be started.
+    clientSocket.emit("start-game");
+  });
 
-        // Subscribe to lobby-code
-        clientSocket.on('lobby-code', (lobbyCodeDTO) => {
-            expect(lobbyCodeDTO.code).toBeDefined();
-            // Only emit reset-lobby once lobby code has been received
-            clientSocket.emit('reset-lobby');
-        });
+  test("host starts game event", (done) => {
+    let createLobbyDTO = new CreateLobbyDTO("Anmol");
 
-        // Finish test once reset-lobby-update has been received
-        clientSocket.on('reset-lobby-update', () => {
-            done();
-        });
+    // Request to create a new lobby
+    // this will set the socket id to the host to be tested.
+    clientSocket.emit("create-lobby", createLobbyDTO);
 
-        // Request to create a new lobby
-        clientSocket.emit('create-lobby', createLobbyDTO);
+    // Host client should subscribe to confirm-game-start - this is the host client.
+    clientSocket.on("confirm-game-start", () => {
+      // Host should receive a different event than the client
+      done();
     });
+
+    // todo: doesnt work until justin pushes his code in.
+    // // Or Client should recieve a to game-ready - this is the non-host client.
+    // clientSocket.on("game-ready", () => {
+    //     // Host should receive a different event than the client
+    //     done();
+    // });
+
+    // Request to server that lobby is full.
+    clientSocket.emit("lobby-ready", createLobbyDTO);
```

Comment:
Think the server should emit to the client that the lobby is ready such that the client can enable the button to start the game. 

Under this implementation it is the client that is telling the server that the lobby is ready which I think is incorrect

Reply:
i deleted the event handler